test/plugin/test_out_google_cloud.rb

```diff
@@ -742,9 +742,8 @@ def test_timestamps
     [Time.at(123_456.789), Time.at(0), Time.now].each do |ts|
       # Test the "native" fluentd timestamp as well as our nanosecond tags.
       d.emit({ 'message' => log_entry(emit_index) }, ts.to_f)
-      # The native timestamp currently only supports second granularity
-      # (fluentd issue #461), so strip nanoseconds from the expected value.
-      expected_ts.push(Time.at(ts.tv_sec))
+
+      expected_ts.push(Time.at(ts))
```

Comment:
This is just `expected_ts.push(ts)`. Once you do this, it shouldn't be flaky, but please double-check.